src/sentry/plugins/sentry_mail/activity/release.py

```diff
@@ -100,20 +100,38 @@ def get_participants(self):
         if not self.email_list:
             return {}
 
-        # identify members which have been seen in the commit log and have
-        # verified the matching email address
-        return {
-            user: GroupSubscriptionReason.committed
+        # collect all users with verified emails on a team in the related projects,
+        # and couple them with the the users' setting value for deploy-emails
+        users_with_options = [
```

Comment:
so maybe it's not a big deal because i think the `__metadata` is a caching mechanism, but you could probably just get the options for all users at the same time by doing something like:
```python
users = list(User.objects.filter(
    emails__is_verified=True,
    sentry_orgmember_set__teams=Team.objects.filter(
        id__in=[p.team_id for p in self.projects]
    ),
    is_active=True,
).distinct())

options_by_user_id = {
    uo.user_id: uo 
    for uo in UserOption.objects.filter(
        user__in=users,
        organization=self.organization,
        key='deploy-emails',
    )
}

users_with_options = [
    (user, options_by_user_id.get(user.id, UserOptionValue.committed_deploys_only))
    for user in users
]
```
it kinda seems like the caching is optimized for getting all results for a single user, so maybe something like that could improve performance? not totally sure...